flink/src/main/java/org/apache/iceberg/flink/FlinkTypeToType.java

```diff
@@ -171,6 +171,7 @@ public Type visit(MapType mapType) {
   }
 
   @Override
+  @SuppressWarnings("ReferenceEquality")
```

Comment:
this is for the `boolean isRoot = root == rowType` check, which seems to be on purpose, but maybe you could double check whether using ref. equality here is still wanted? Same for `SparkTypeToType`